main.go

```diff
@@ -177,8 +177,9 @@ func main() {
 	}
 
 	if err = (&policysetctrl.PolicySetReconciler{
-		Client: mgr.GetClient(),
-		Scheme: mgr.GetScheme(),
+		Client:   mgr.GetClient(),
+		Scheme:   mgr.GetScheme(),
+		Recorder: mgr.GetEventRecorderFor(policysetctrl.ControllerName),
```

Comment:
Recorder is not being used anywhere. I suggest to generate events for policyset CR change.